.github/workflows/config/spellcheck_config.yml

```diff
@@ -157,11 +157,14 @@ matrix:
       # Ignore content between inline backticks
       - open: '(?P<open>`+)'
         close: '(?P=open)'
-      # Ignore the first lowercase word after @param, which is usually the parameter name
-      - open: '@param(?:\s[a-z]+)?'
+      # Ignore the first word after \command or @command, which is a code artifact
+      - open: '[\\@]([abcep]|em)(?:\s\S+)?'
         close: '(\s|$)'
-      # Ignore words that start with an @, e.g. @param used in doc comments
-      - open: '@\S'
+      # Ignore the doxygen \param command and the code artifact after it.
+      - open: '[\\@]param(\s+\[[a-z]+\])?\s+(?:\S+)'
+        close: '(\s|$)'
+      # Ignore words that start with a \ or @, e.g. @param used in doc comments
+      - open: '[\\@](\S+|[n\\@&\$#<>%"\.=|]|::|--|---)'
```

Comment:
I'm not sure I follow what `[n\\@&\$#<>%"\.=|]|::|--|---` is trying to catch, but I believe the following complete changes would work for you.

From:
```
      # Ignore the first lowercase word after @param, which is usually the parameter name
      - open: '@param(?:\s[a-z]+)?'
```
To:
```
      # Ignore the first word after @param, which is usually the parameter name
      - open: '@param(?:\s\w+)?'
        close: '(\s|$)'
      # Ignore the first word after \p, which is usually the parameter name
      - open: '\\p(?:\s\w+)?'
        close: '(\s|$)'
```

Reply:
The regular expression is not trying to merely allow `\p` and `\param`. It is allowing any of the multitude of valid doxygen commands as can be found [in the documentation](https://www.doxygen.nl/manual/commands.html). The spellchecking script is too restrictive and rejects the use of valid doxygen which discourages writing documentation.